dask/diagnostics/profile_visualize.py

```diff
@@ -103,7 +104,9 @@ def visualize(results, dsk, palette='GnBu', file_path="profile.html",
     The completed bokeh plot object.
     """
 
-    bp.output_file(file_path)
+    if not _state._notebook:
```

Comment:
Per talking to @bryevdv earlier today, while not perfect, this isn't the worst thing ever. `_state` isn't external api, but it's also not likely to change. The other option is to have users always call `ouput_file` or `output_notebook` themselves. I kind of like being able to just call `prof.visualize()` and have everything work, but could go both ways.